core/services/wifi/Hotspot.py

```diff
@@ -0,0 +1,339 @@
+import pathlib
+import shlex
+import shutil
+import subprocess
+import tempfile
+import time
+from enum import Enum
+from ipaddress import IPv4Address
+from typing import Any, List, Optional
+
+import psutil
+from commonwealth.utils.DHCPServerManager import Dnsmasq as DHCPServerManager
+from loguru import logger
+from pyroute2 import IW, IPRoute
+
+
+class HostapdFrequency(str, Enum):
+    """Valid hostapd frequency modes."""
+
+    HW_2_4 = "g"  # Hostapd id for 2.4 GHz mode
+    HW_5_0 = "a"  # Hostapd id for 5.0 GHz mode
+
+    @staticmethod
+    def mode_from_channel_frequency(channel: int) -> "HostapdFrequency":
+        return HostapdFrequency.HW_2_4 if str(channel).startswith("2") else HostapdFrequency.HW_5_0
```

Comment:
The hostapd input is the channel, not the frequency.

Reply:
Ok, can you specify what are the inputs of channel ?
Are they under the 24XXMHz range ? Just to understand more the `str(channel).startswith("2")` logic, since it appears a bit weird looking only at the code and what it tried to accomplish.

Reply:
They are. In this case, the method argument had the wrong name. Fixed that. Changed the condition to use the `int(...) == 24`, although I imagine we won't have 2.5-2.9GHz networks anytime soon.